nornir/core/__init__.py

```diff
@@ -168,6 +169,13 @@ def close_connections_task(task):
 
         self.run(task=close_connections_task, on_good=on_good, on_failed=on_failed)
 
+    @property
+    def _runner(self) -> RunnerPlugin:
```

Comment:
shouldn't we do the opposite? store in `_runner` and have the propery be `runner`?

Reply:
I tried that first however it breaks some other things, in the below functions we use `self.__dict__` which causes issues with an attributed such as `self._runner` as we'd send in "_runner" as an argument to Nornir instead of "runner".

```python
    def with_processors(self, processors: List[Processor]) -> "Nornir":
        """
        Given a list of Processor objects return a copy of the nornir object with the processors
        assigned to the copy. The orinal object is left unmodified.
        """
        return Nornir(**{**self.__dict__, **{"processors": Processors(processors)}})

    def with_runner(self, runner: RunnerPlugin) -> "Nornir":
        """
        Given a runner return a copy of the nornir object with the runner
        assigned to the copy. The orinal object is left unmodified.
        """
        return Nornir(**{**self.__dict__, **{"runner": runner}})

    def filter(self, *args: Any, **kwargs: Any) -> "Nornir":
        """
        See :py:meth:`nornir.core.inventory.Inventory.filter`

        Returns:
            :obj:`Nornir`: A new object with same configuration as ``self`` but filtered inventory.
        """
        b = Nornir(**self.__dict__)
        b.inventory = self.inventory.filter(*args, **kwargs)
        return b
```

What we could do instead is to replace `self.__dict__` in these cases with a property that returns a dictionary where the keys are hardcoded instead, i.e: inventory, data, config, runner, processors.

Reply:
yeah, either that or just a cleaner clone method